### PR TITLE
PostgreSQL requirements updates

### PR DESCRIPTION
**Pull request by sudomateo**
Detailed the new automatic schema creation feature and updated the documentation to be a bit clearer.

### CODE REVIEW

File: content/source/docs/enterprise/before-installing/postgres-requirements.html.md [hunk 1]
```diff
@@ -7,37 +7,41 @@ page_title: "PostgreSQL Requirements - Before Installing - Terraform Enterprise"
...
-When Terraform Enterprise uses an external PostgreSQL database, the
-following must be present on it:
+To use an external PostgreSQL database with Terraform Enterprise, the following
+requirements must be met:
 
-* PostgreSQL version `>= 9.4` and `< 10.0`
-* User with the ability to create/modify/read tables and indices on all schemas created
-  * If it's not feasible to have a user with "CREATE EXTENSION", then create the [extensions](#extensions) below before installation
-* The following PostgreSQL schemas must be installed into the database: `rails`, `vault`, `registry`
+* The PostgreSQL Server version must be `>= 9.4` and `< 10.0`.
```
Comment (evanphx) [inline content/source/docs/enterprise/before-installing/postgres-requirements.html.md]: This can be relaxed to `< 11.0`

Comment (sudomateo) [reply content/source/docs/enterprise/before-installing/postgres-requirements.html.md]: Done!

File: content/source/docs/enterprise/before-installing/postgres-requirements.html.md [hunk 2]
```diff
@@ -7,37 +7,41 @@ page_title: "PostgreSQL Requirements - Before Installing - Terraform Enterprise"
...
-When Terraform Enterprise uses an external PostgreSQL database, the
-following must be present on it:
+To use an external PostgreSQL database with Terraform Enterprise, the following
+requirements must be met:
 
-* PostgreSQL version `>= 9.4` and `< 10.0`
-* User with the ability to create/modify/read tables and indices on all schemas created
-  * If it's not feasible to have a user with "CREATE EXTENSION", then create the [extensions](#extensions) below before installation
-* The following PostgreSQL schemas must be installed into the database: `rails`, `vault`, `registry`
+* The PostgreSQL Server version must be `>= 9.4` and `< 11.0`.
```
Comment (glasner) [inline content/source/docs/enterprise/before-installing/postgres-requirements.html.md]: Not sure if we're ready to bump the version to include 10?

File: content/source/docs/enterprise/before-installing/postgres-requirements.html.md [hunk 3]
```diff
@@ -7,37 +7,41 @@ page_title: "PostgreSQL Requirements - Before Installing - Terraform Enterprise"
...
+* A PostgreSQL user must be created with the following permissions on the database:
+  * The ability to create, modify, and read all tables and indices on all schemas within the database. Usually this is granted if the user is an owner of the database.
+  * The ability to create extensions. If it is not feasible to have a user with the "CREATE EXTENSION" privilege, then refer to the [Creating Extensions](#creating-extensions) section below for information on creating the necessary extensions.
+* The `rails`, `vault`, and `registry` PostgreSQL schemas must be created on the database. These schemas will be automatically created by the Terraform Enterprise application if they do not already exist. To create the schemas manually, refer to the [Creating Schemas](#creating-schemas) section below.
```
Comment (glasner) [inline content/source/docs/enterprise/before-installing/postgres-requirements.html.md]: Schema parts reads oddly now. Says "schemas must exist" then next sentence says they "will be automatically created".

Do we even need to talk about schemas anymore here?

File: content/source/docs/enterprise/before-installing/postgres-requirements.html.md [hunk 4]
```diff
@@ -7,37 +7,41 @@ page_title: "PostgreSQL Requirements - Before Installing - Terraform Enterprise"
...
 ```sql
 CREATE SCHEMA rails;
 CREATE SCHEMA vault;
 CREATE SCHEMA registry;
 ```
 
-When providing optional extra keyword parameters for the database connection,
-note an additional restriction on the `sslmode` parameter is that only the
-`require`, `verify-full`, `verify-ca`, and `disable` values are allowed.
-
```
Comment (glasner) [inline content/source/docs/enterprise/before-installing/postgres-requirements.html.md]: Should we be deleting this?

File: content/source/docs/enterprise/before-installing/postgres-requirements.html.md [hunk 5]
```diff
@@ -7,37 +7,41 @@ page_title: "PostgreSQL Requirements - Before Installing - Terraform Enterprise"
...
-  * If it's not feasible to have a user with "CREATE EXTENSION", then create the [extensions](#extensions) below before installation
-* The following PostgreSQL schemas must be installed into the database: `rails`, `vault`, `registry`
+* The PostgreSQL Server version must be `>= 9.4` and `< 11.0`.
+* A PostgreSQL user must be created with the following permissions on the database:
+  * The ability to create, modify, and read all tables and indices on all schemas within the database. Usually this is granted if the user is an owner of the database.
+  * The ability to create extensions. If it is not feasible to have a user with the "CREATE EXTENSION" privilege, then refer to the [Creating Extensions](#creating-extensions) section below for information on creating the necessary extensions.
+* The `rails`, `vault`, and `registry` PostgreSQL schemas must be created on the database. These schemas will be automatically created by the Terraform Enterprise application if they do not already exist. To create the schemas manually, refer to the [Creating Schemas](#creating-schemas) section below.
 
 ## Creating Schemas
```
Comment (glasner) [inline content/source/docs/enterprise/before-installing/postgres-requirements.html.md]: I would change this section name to "Manually Creating Schemas" and then add a note about this is only required in versions below X

File: content/source/docs/enterprise/before-installing/postgres-requirements.html.md (hunk 3 above)
Comment (glasner) [reply content/source/docs/enterprise/before-installing/postgres-requirements.html.md]: To update.. maybe we do need to keep but just as note about version earlier than X needing to click to view section below about creating schemas?

File: content/source/docs/enterprise/before-installing/postgres-requirements.html.md (hunk 2 above)
Comment (sudomateo) [reply content/source/docs/enterprise/before-installing/postgres-requirements.html.md]: I did that based on previous input from @evanphx.

File: content/source/docs/enterprise/before-installing/postgres-requirements.html.md (hunk 3 above)
Comment (sudomateo) [reply content/source/docs/enterprise/before-installing/postgres-requirements.html.md]: I'll update this to include a note about the change in behavior starting at version X.

File: content/source/docs/enterprise/before-installing/postgres-requirements.html.md (hunk 4 above)
Comment (sudomateo) [reply content/source/docs/enterprise/before-installing/postgres-requirements.html.md]: I didn't delete it. I moved it to the bottom of the file under a new section.

File: content/source/docs/enterprise/before-installing/postgres-requirements.html.md (hunk 5 above)
Comment (sudomateo) [reply content/source/docs/enterprise/before-installing/postgres-requirements.html.md]: Will do!

File: content/source/docs/enterprise/before-installing/postgres-requirements.html.md (hunk 2 above)
Comment (glasner) [reply content/source/docs/enterprise/before-installing/postgres-requirements.html.md]: Cool, nice to hear 🙌

File: content/source/docs/enterprise/before-installing/postgres-requirements.html.md (hunk 4 above)
Comment (glasner) [reply content/source/docs/enterprise/before-installing/postgres-requirements.html.md]: Oh missed that. Thanks

**Pull request merged**